packages/next/src/server/app-render/make-get-server-inserted-html.tsx

```diff
@@ -81,6 +84,14 @@ export function makeGetServerInsertedHTML({
               return <script key={polyfill.src} {...polyfill} />
             })
         }
+        {traceData.map(({ key, value }) => {
```

Comment:
I am not fully sure if I understand your concern correctly, but I think this is already supported.

If you have multiple OpenTelemetry propagation keys that have the same value, they will already be added as multiple `<meta />` tags. [Here we generate the array](https://github.com/vercel/next.js/pull/64256/files#diff-27f951b8613914f9b7e5a8a8a8bbf4a151bf1202f7e290a46e5ed5480567f138R187-R192), and right where you commented, we render the tags.

If your comment is about having additional attributes on the rendered `<meta />` tags, I don't think that is needed because OpenTelemetry trace data is just key-value pairs.

Reply:
I was checking what will the `getTracePropagationData` can be used, seems not all of them will be used for rendering injected metadata. Some could be used in request/response headers. Maybe we should give everything a prefix for categorizing. Such as `<category>@<key>:<valye>`

To inject metadata rendering tags we can use  `metadata@meta-tag-key:the-value` for identify it's for metadata and we only filter the ones starting with `metadata@` for injecting meta tags.

If we need to use propagation data for other cases, we can add a different category, such as `headers@key:value`

Reply:
Trying to sum up my thoughts (and also our slack convo):

Long story short, I think adding something like a `namespace@` would not work for the use case I was attempting to fix with this PR and would be mixing too many concepts.

Generally speaking, OpenTelemetry users have a convention for the name of the keys in their propagation data. Having a `metadata@` prefix would mess with that convention. When implementing an OpenTelemetry `Propagator`, you cannot know at runtime whether your data will be `inject`ed into a meta tag or attached as headers to outgoing requests. Having the `metadata@` prefix in headers of outgoing requests would certainly be problematic in many cases.

I do agree with the fact that Next.js probably needs a way to dynamically attach response headers. However, I don't think using OpenTelemetry is the right vehicle for that. Returning trace information to the client as response headers is generally of no use, because you cannot extract it with JS (except for very few exceptions like `Server-Timing`). I suggest that if you guys want to add a mechanism to return response headers you add it through a more explicit API.